librz/analysis/arch/x86/x86_il.c

```diff
@@ -0,0 +1,3806 @@
+		RzILOpEffect *set_dest = SETL("_dest", UNSIGNED(ins->structure->operands[0].size * BITS_PER_BYTE, VARL("_tmp_xp")));
+		RzILOpEffect *set_operand = x86_il_set_op(0, VARL("_dest"));
+
+		/* Check if the result fits in the destination */
+		RzILOpPure *cond = EQ(SIGNED(ins->structure->operands[0].size * 2 * BITS_PER_BYTE, VARL("_dest")), VARL("_tmp_xp"));
+		RzILOpEffect *true_branch = SEQ2(SETG(EFLAGS(CF), IL_FALSE), SETG(EFLAGS(OF), IL_FALSE));
+		RzILOpEffect *false_branch = SEQ2(SETG(EFLAGS(CF), IL_TRUE), SETG(EFLAGS(OF), IL_TRUE));
+
+		return SEQ5(dest, tmp_xp, set_dest, set_operand, BRANCH(cond, true_branch, false_branch));
+	}
+	case 3: {
+		RzILOpEffect *tmp_xp = SETL("_tmp_xp", MUL(SIGNED(ins->structure->operands[1].size * 2 * BITS_PER_BYTE, x86_il_get_op(1)), SIGNED(ins->structure->operands[1].size * 2 * BITS_PER_BYTE, x86_il_get_op(2))));
+		RzILOpEffect *set_dest = SETL("_dest", UNSIGNED(ins->structure->operands[0].size * BITS_PER_BYTE, VARL("_tmp_xp")));
+		RzILOpEffect *set_operand = x86_il_set_op(0, VARL("_dest"));
+
+		/* Check if the result fits in the destination */
+		RzILOpPure *cond = EQ(SIGNED(ins->structure->operands[0].size * 2 * BITS_PER_BYTE, VARL("_dest")), VARL("_tmp_xp"));
+		RzILOpEffect *true_branch = SEQ2(SETG(EFLAGS(CF), IL_FALSE), SETG(EFLAGS(OF), IL_FALSE));
+		RzILOpEffect *false_branch = SEQ2(SETG(EFLAGS(CF), IL_TRUE), SETG(EFLAGS(OF), IL_TRUE));
+
+		return SEQ4(tmp_xp, set_dest, set_operand, BRANCH(cond, true_branch, false_branch));
+	}
+	default:
+		RZ_LOG_ERROR("RzIL: x86: IMUL: Invalid operand count\n");
+		return NULL;
+	}
+
+	return NULL;
+}
+
+// /**
+//  * IN
+//  * Input from port
+//  * Encodings: I, ZO
+//  */
+// IL_LIFTER(in) {
+// 	/* TODO: Implement after I/O ports acccessing implemented in IL */
+// 	return EMPTY();
+// }
+
+/**
+ * INC
+ * Increment by 1
+ * Encodings: M, O
+ */
+IL_LIFTER(inc) {
+	RzILOpEffect *op = SETL("_op", x86_il_get_op(0));
+	RzILOpEffect *result = SETL("_result", ADD(VARL("_op"), UN(ins->structure->operands[0].size * BITS_PER_BYTE, 1)));
+	RzILOpEffect *set_result = x86_il_set_op(0, VARL("_result"));
+
+	RzILOpEffect *arith_flags = x86_il_set_arithmetic_flags_except_cf(VARL("_result"), VARL("_op"), UN(ins->structure->operands[0].size * BITS_PER_BYTE, 1), true);
+	RzILOpEffect *res_flags = x86_il_set_result_flags(VARL("_result"));
+
+	return SEQ5(op, result, set_result, arith_flags, res_flags);
+}
+
+/**
+ * INT
+ * Call to interrupt procedure
+ * Encodings: I, ZO
+ */
+IL_LIFTER(int) {
+	/* For now, it just jumps to an empty goto label "int"
+	Also need an EMPTY() instruction after it to denote
+	the end of analysis and restart all IL analysis */
+	return SEQ2(GOTO("int"), EMPTY());
+}
+
+static void label_int(RzILVM *vm, RzILOpEffect *op) {
+	// empty "int" label
+	return;
+}
+
+/**
+ * INTO
+ * Call to interrupt if overflow flag set
+ */
+IL_LIFTER(into) {
+	return BRANCH(VARG(EFLAGS(OF)), SEQ2(GOTO("int"), EMPTY()), NOP());
+}
+
+// /**
```

Comment:
I am not 100% sure, but we could uplift it the similar way as `RET` opcode.

Reply:
Not really. Because what gets popped depends on the CPU state and there are a bunch of other sorts of checks and conditions (because it's an interrupt return ¯\_(ツ)_/¯). Majority of them can be ignored, since we only care about emulation. Also, I am pretty sure that it is very very hard to use this instruction in any userland binary.